common/compiler/solidity.go

```diff
@@ -54,6 +49,15 @@ var (
 		"--add-std",
 		"1",
 	}
+	paramsNew = []string{
+		"--bin",        // Request to output the contract in binary (hexadecimal).
+		"--abi",        // Request to output the contract's JSON ABI interface.
+		"--userdoc",    // Request to output the contract's Natspec user documentation.
+		"--devdoc",     // Request to output the contract's Natspec developer documentation.
+		"--add-std",    // include standard lib contracts
+		"--optimize=1", // code optimizer switched on
```

Comment:
I just rebuit geth develop branch with this fix and I think @chriseth is correct that this does NOT work...
```
"solc: exit status 1
option '--optimize' does not take any arguments"
```